pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -713,6 +813,17 @@ def _is_valid_type(
     if expected_type is UNDEFINED:
         return True
 
+    # Special case for device_class
+    if expected_type == DEVICE_CLASS and in_return:
+        return (
+            isinstance(node, nodes.Name)
+            and node.name
+            and node.name.endswith("DeviceClass")
+            or isinstance(node, nodes.Attribute)
+            and node.attrname
+            and node.attrname.endswith("DeviceClass")
```

Comment:
```suggestion
            isinstance(node, nodes.Name)
            and node.name.endswith("DeviceClass")
            or isinstance(node, nodes.Attribute)
            and node.attrname.endswith("DeviceClass")
```
AFAIK you can assume that `name` and `attrname` will always be a string. The typing in astroid is not ideal there. _Basically it could theoretically be initialized with `None`, but isn't ever. We just need to deprecate it probably sometime._

Reply:
I assumed it would cause issues for mypy - but apparently mypy is happy.
I've adjusted accordingly.

Reply:
Astroid (and pylint) doesn't have a `py.typed` file yet. Thus mypy just ignores them.